extract common functions and implemente mwan3rule

Bigzhuang opened the pull request:
Change-Id: Ief9ab8094ab249393a854805622f51fce064b101

cheng1li committed:
extract common functions and implemente mwan3rule

Change-Id: Ief9ab8094ab249393a854805622f51fce064b101

cheng1li commented on platform/crd-ctrlr/src/deployement.yaml:
```diff
@@ -0,0 +1,295 @@
+apiVersion: v1
```
Is this file for the debug/test?

Bigzhuang replied:
yes, just for debug

cheng1li replied:
Let's remove this yaml file

xuyizhox committed:
Update develope env to k8s 1.16

1. make crd function work properly on k8s 1.16
2. replace parameter in process_reconcil with global var

cheng1li commented on platform/crd-ctrlr/src/controllers/mwan3policy_controller.go:
```diff
@@ -17,19 +17,110 @@ package controllers
 	"github.com/go-logr/logr"
-	"k8s.io/apimachinery/pkg/api/errors"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	// "k8s.io/apimachinery/pkg/api/errors"
+	// metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
```
Instead of commenting out these lines, let's remove them directly. As well as the other comments

cheng1li reviewed (commented):
Did you get a chance to test running controller as pod more than `make run`?

Bigzhuang reviewed (commented):
> Did you get a chance to test running controller as pod more than `make run`?

haven't test controller as pod by now, i'd like to add it to my TODO list

xuyizhox committed:
Add Message to SdewanStatuswirte error msg to Message field in SdewanStatus, to get applied failed cr reason detail

from my perspective by now, CR may get Error message in below cases:
1. get errors due to desired cnf pods not exist (code added & tested)
2. get errors due to dependent cr not exist,like Mwan3policy to Mwan3rule(TBD)

cheng1li commented:
@Bigzhuang , could  you please add a comment to indicate what's the state of your latest commit? What tests have you done against it?

Bigzhuang commented:
> @Bigzhuang , could you please add a comment to indicate what's the state of your latest commit? What tests have you done against it?

sure
